### PR TITLE
Add the French translation

### PR DESCRIPTION
**Pull request by QuantumSheep**
Damn it, that took me a lot of time! 

I start from the principle that improvements can be done to this traduction and that it's not perfect (but almost 😏). By the way, I will improve it along the time and make updates to follow your book's ones.

### CODE REVIEW

**Comment by bagder** [comment]
Lovely. You also want to add 'fr' to `LANGS.md` in the top directory.

**Comment by bagder** [comment]
And feel free to add yourself here: https://github.com/bagder/http3-explained/blob/master/HOWTO-TRANSLATE.md#main-translators

File: LANGS.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
 * [English](en/)
+* [French](fr/)
```
Comment (bagder) [inline LANGS.md]: Please use the native name there, "Français" I presume?

**Comment by QuantumSheep** [comment]
For the `LANGS.md` file, I put French in the alphabetical order (to explain why it's in second)

**Pull request merged**